compile error with rust 1.82 and "Bump thiserror from 1.0.69 to 2.0.3 (#917)"

### Component

- [ ] CLI
- [x] SDK
- [ ] Something else
- [ ] Not sure

## What was the expected behaviour

compile works

## What is the current behaviour and what actions did you take to get there

**NOTE** @iximeow has said they can only get this to happen when `cargo add`-ing oxide.rs as a dependency to another crate. So far this has been my experience as well, even after updating the rust-toolchain.toml to 1.82. Which might explain why CI has not had an issue.

### This succeeds with rust 1.79, but fails with rust 1.82:

Using commit 33b1cbc90eeceb07fdfbe3d02d5b24cce67aa11e which bumps thiserror. Any commit after this, up to the latest commit, exhibits this problem.

```
$ cargo init
$ cargo add --git https://github.com/oxidecomputer/oxide.rs/ oxide --rev 33b1cbc90eeceb07fdfbe3d02d5b24cce67aa11e
$ cargo check
```

FULL LOGS:
- pass (1.79) - [pass-1.79.txt](https://github.com/user-attachments/files/17912892/pass-1.79.txt)
- fail (1.82) - [fail-1.82.txt](https://github.com/user-attachments/files/17912897/fail-1.82.txt)

I have not tested with rust 1.80 or 1.81.

### This succeeds with rust 1.82:

Note this is the commit right before the thiserror bump

```
$ cargo init
$ cargo add --git https://github.com/oxidecomputer/oxide.rs/ oxide --rev af7502b341454bed4ff425039dd037d7113ad4e2
$ cargo check
```

FULL LOGS:
- pass (1.82) - [pass-thiserror-1.0.69.txt](https://github.com/user-attachments/files/17912898/pass-thiserror-1.0.69.txt)


### When it fails, this is the error message

```
error[E0599]: the method `as_display` exists for reference `&PathBuf`, but its trait bounds were not satisfied
  --> /home/seraphim/.cargo/git/checkouts/oxide.rs-96e27eb02ef04370/33b1cbc/sdk/src/lib.rs:33:13
   |
33 |     #[error("Parse error for {0}: {1}")]
   |             ^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: the following trait bounds were not satisfied:
           `PathBuf: std::fmt::Display`
           which is required by `&PathBuf: AsDisplay<'_>`

error[E0599]: the method `as_display` exists for reference `&PathBuf`, but its trait bounds were not satisfied
  --> /home/seraphim/.cargo/git/checkouts/oxide.rs-96e27eb02ef04370/33b1cbc/sdk/src/lib.rs:39:13
   |
39 |     #[error("Profile information not present in {0} for {1}")]
   |             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: the following trait bounds were not satisfied:
           `PathBuf: std::fmt::Display`
           which is required by `&PathBuf: AsDisplay<'_>`
```

## CLI or SDK version

33b1cbc90eeceb07fdfbe3d02d5b24cce67aa11e and newer

## Operating system

Linux

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -53,7 +53,7 @@ serde = { version = "1.0.215", features = ["derive"] }
 serde_json = "1.0.133"
 similar = "2.6.0"
 tabwriter = "1.4.0"
-thiserror = { version = "2", default-features = false }
+thiserror = "2.0.3"
 tempfile = "3.14.0"
 test-common = { path = "test-common" }
 thouart = { git = "https://github.com/oxidecomputer/thouart" }

--- a/rust-toolchain.toml
+++ b/rust-toolchain.toml
@@ -1,3 +1,3 @@
 [toolchain]
-channel = "1.80"
+channel = "1.82"
 profile = "default"

